extensions/jdbc/jdbc-db2/deployment/src/main/java/io/quarkus/jdbc/db2/deployment/JDBCDB2Processor.java

```diff
@@ -51,6 +54,15 @@ void configureAgroalConnection(BuildProducer<AdditionalBeanBuildItem> additional
         }
     }
 
+    @BuildStep
+    void registerDriverForReflection(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
+        //Not strictly necessary when using Agroal, as it also registers
+        //any JDBC driver being configured explicitly through its configuration.
+        //We register it for the sake of people not using Agroal,
+        //for example when the driver is used with OpenTelemetry JDBC instrumentation.
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, DB2_DRIVER_CLASS));
+    }
+
```

Comment:
I don't understand that part. Because I'm pretty sure that would be a problem for all the drivers around? Or is Db2 specific in this regard?

Reply:
Adding this registering for reflection fixes the issue with OT as then Agroal does not register driver itself (it registers OT driver for reflection) and IT I added failed.

Reply:
Okay :).

I think we should probably take a step back and implement OpenTelemetry for JDBC a bit differently by using a similar pattern to what we did in https://github.com/quarkusio/quarkus/pull/26144 for OpenTracing. I have not checked if it's doable but I think we should at least have a look.

@brunobat does it sound like a good idea to you?

In the meantime, let's merge this to fix the issue.

Reply:
At the consistency level it would be preferable and it's also more user friendly.
What do you think @michalvavrik ?

Reply:
created this: https://github.com/quarkusio/quarkus/issues/30532

Reply:
Sorry, I'm very busy these days. I'll look into #30532 .